lib/scala/project-manager/src/main/scala/org/enso/projectmanager/boot/configuration.scala

```diff
@@ -41,21 +44,37 @@ object configuration {
     */
   case class ServerConfig(host: String, port: Int)
 
-  /** A configuration object for properties of project storage.
+  /** A configuration object for metadata storage.
     *
-    * @param projectsRoot a project root
-    * @param userProjectsPath a user project root
-    * @param projectMetadataDirectory a directory name containing project
-    *                                 metadata
+    * @param projectMetadataDirectory a directory name containing project metadata
     * @param projectMetadataFileName a name of project metadata file
     */
-  case class StorageConfig(
-    projectsRoot: File,
-    userProjectsPath: File,
+  case class MetadataStorageConfig(
     projectMetadataDirectory: String,
     projectMetadataFileName: String
   )
 
+  /** A configuration object for properties of project storage.
+    *
+    * @param projectsRoot overrides user projects root directory
+    * @param projectsDirectory a user projects directory
+    * @param metadata a metadata storage config
+    */
+  case class StorageConfig(
+    projectsRoot: Option[File],
+    projectsDirectory: String,
```

Comment:
`File` for `projectRoot` and `String` for a directory? Why not use `File` for both?

Reply:
File values represent paths, and string values represent names